src/main/kotlin/lotto/domain/LottoTicket.kt

```diff
@@ -1,11 +1,17 @@
 package lotto.domain
 
-data class LottoTicket(private val generateLottoNumbers: () -> Set<Int>) {
-    val lottoNumbers = LottoNumbers(generateLottoNumbers().map { LottoNumber.of(it) }.toSet())
+sealed class LottoTicket {
+    abstract val lottoNumbers: LottoNumbers
 
     fun checkLottoWinnerRank(lottoWinnerNumbers: LottoWinnerNumbers): LottoWinnerRank {
         val matchCount = lottoNumbers.checkLottoNumbersMatch(lottoWinnerNumbers.lottoNumbers)
         val bonusCheck = lottoNumbers.contains(lottoWinnerNumbers.bonusNumber)
         return LottoWinnerRank.getRankByMatches(matchCount = matchCount, bonusCheck = bonusCheck)
     }
+
+    class ManualLottoTicket(override val lottoNumbers: LottoNumbers) : LottoTicket()
+
+    class AutoLottoTicket(private val generateLottoNumbers: () -> LottoNumbers) : LottoTicket() {
```

Comment:
generateLottoNumbers는 프로퍼티는 아닌걸로 보이네요! :) 

```suggestion
    class AutoLottoTicket(generateLottoNumbers: () -> LottoNumbers) : LottoTicket() {
```

Reply:
해당 함수는 AutoLottoTicket 생성 시점에만 호출되어 로또 번호를 생성하므로, 프로퍼티 대신 바로 생성자 내부에서 값을 초기화하는 방식으로 수정하는 것이 더 적절하겠네요!

덕분에 불필요한 프로퍼티 선언을 제거하고, 코드의 명확성과 간결성을 높일 수 있을 것 같습니다. 

제안해주신대로 변경했습니다 👍